File: 75 days Challenge Leetcode/HashMap+Set/Day-20/Solution.java
import java.util.*;

class Solution {
    public List<List<Integer>> findDifference(List<Integer> nums1, List<Integer> nums2) {
        // to eliminate duplicates
        Set<Integer> set1 = new HashSet<>(nums1);
        Set<Integer> set2 = new HashSet<>(nums2);
        
        // to store unique elements for each input list
        List<Integer> dp = new ArrayList<>();
        List<Integer> dp1 = new ArrayList<>();
        
        // Find elements that are in set1 but not in set2
        for (int num : set1) {
            if (!set2.contains(num)) {
                dp.add(num); // Add to the result list if the element is unique to set1
            }
        }

        // Find elements that are in set2 but not in set1
        for (int num : set2) {
            if (!set1.contains(num)) {
                dp1.add(num); // Add to the result list if the element is unique to set2
            }
        }

        return Arrays.asList(dp, dp1);
    }
}

File: 75 days Challenge Leetcode/HashMap+Set/Day-20/Solution.py
from typing import List, Set

class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        # to eliminate duplicates
        set1: Set[int] = set(nums1)
        set2: Set[int] = set(nums2)
        
        # Initialize lists to store unique elements for each input list
        dp = []
        dp1 = []

        # Find elements that are in set1 but not in set2
        for num in set1:
            if num not in set2:
                dp.append(num) # Add to result if unique to set1

        # Find elements that are in set2 but not in set1
        for num in set2:
            if num not in set1:
                dp1.append(num) # Add to result if unique to set2

        # Return a list of lists containing the unique elements for each input list
        return [dp, dp1]
